api-flask/app/main.py

```diff
@@ -67,84 +91,88 @@ def _calculate_stats(zones,
+
+        if geotiff_url is None:
             logger.error('Received {}'.format(data))
-            err_message = '{} required within wfs_params object'
-            joined_missing = ','.join(missing)
+            raise BadRequest('geotiff_url is required.')
 
-            raise BadRequest(err_message.format(joined_missing))
+        if zones_geojson is None and zones_url is None:
+            logger.error('Received {}'.format(data))
+            raise BadRequest('One of zones or zones_url is required.')
 
-        wfs_response = get_wfs_response(wfs_params)
+        geojson_out = strtobool(data.get('geojson_out', 'False'))
+        group_by = data.get('group_by')
 
-    intersect_comparison = None
-    if intersect_comparison_string is not None:
-        intersect_comparison = validate_intersect_parameter(
-            intersect_comparison_string
+        geotiff = cache_file(
+            prefix='raster',
+            url=geotiff_url,
+            extension='tif'
         )
 
-    features = _calculate_stats(
-        zones,
-        geotiff,
-        stats=['min', 'max', 'mean', 'median', 'sum', 'std'],
-        prefix='stats_',
-        group_by=group_by,
-        geojson_out=geojson_out,
-        wfs_response=wfs_response,
-        intersect_comparison=intersect_comparison
-    )
+        # TODO - Add validation for zones.
+        if (zones_geojson is not None):
+            zones = cache_geojson(
+                prefix='zones_geojson',
+                geojson=zones_geojson
+            )
+        else:
+            zones = cache_file(
+                prefix='zones',
+                url=zones_url,
+                extension='json',
+            )
+
+        wfs_params = data.get('wfs_params', None)
+
+        wfs_response = None
+        if wfs_params is not None:
+            # Validate required keys.
+            required_keys = ['layer_name', 'url', 'time']
+            missing = [f for f in required_keys if f not in wfs_params.keys()]
+
+            if len(missing) > 0:
+                logger.error('Received {}'.format(data))
+                err_message = '{} required within wfs_params object'
+                joined_missing = ','.join(missing)
+
+                raise BadRequest(err_message.format(joined_missing))
+
+            wfs_response = get_wfs_response(wfs_params)
+
+        intersect_comparison = None
+        if intersect_comparison_string is not None:
+            intersect_comparison = validate_intersect_parameter(
+                intersect_comparison_string
+            )
+
+        features = _calculate_stats(
+            zones,
+            geotiff,
+            stats=['min', 'max', 'mean', 'median', 'sum', 'std'],
+            prefix='stats_',
+            group_by=group_by,
+            geojson_out=geojson_out,
+            wfs_response=wfs_response,
+            intersect_comparison=intersect_comparison
+        )
 
-    return jsonify(features)
+        return features
```

Comment:
Flask RESTX takes care of it automatically. Whatever iterator you send, it will first convert it into a JSON object and then send it across so we don't have to explicitly call `jsonify` :)